processing/src/main/java/org/apache/druid/segment/join/JoinConditionAnalysis.java

```diff
@@ -176,6 +179,14 @@ public boolean canHashJoin()
     return canHashJoin;
   }
 
+  /**
+   * Returns the distinct column keys from the RHS required to evaluate the equi conditions.
+   */
+  public List<String> getRightKeyColumns()
+  {
```

Comment:
I'm not sure if it's clear that this method only applies to the equi-conditions. The javadoc explains it but the method name might be misleading? What do you think?

I don't know if I have a good solution, btw. `getRightKeyColumnsForEquiConditions` is clear but quite a mouthful. Maybe `getRightEquiConditionKeys`.

Reply:
Btw, a Set may be more appropriate than a List here.

Reply:
I like the suggestion. Done.